includes/Core/Util/Reset.php

```diff
@@ -278,4 +281,20 @@ private function handle_reset_action( $nonce ) {
 		);
 		exit;
 	}
+
+	/**
+	 * Performs hard reset if it is enabled programmatically.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function maybe_hard_reset() {
+		$hard_reset_enabled = apply_filters( 'googlesitekit_hard_reset_enabled', false );
```

Comment:
Let's add documentation to this filter like we do elsewhere. How about:
```suggestion
		/**
		 * Filters the hard reset option, which is `false` by default.
		 *
		 * By default, when Site Kit is reset it does not delete "persistent" data
		 * (options prefixed with `googlesitekitpersistent_`). If this filter returns `true`,
		 * all options belonging to Site Kit, including those with the above "persistent"
		 * prefix, will be deleted.
		 *
		 * @since n.e.x.t
		 *
		 * @param bool $hard_reset_enabled If a hard reset is enabled. `false` by default.
		 */
		$hard_reset_enabled = apply_filters( 'googlesitekit_hard_reset_enabled', false );
```

Reply:
Thanks, @tofumatt. Looks good to me.